### PR TITLE
Add gesture - poweroff and restart device

### PR DESCRIPTION
**Pull request by robert00s**
Ref: #5021 

Two new gesture:
+ Power off
+ Reboot the device

### CODE REVIEW

File: frontend/apps/reader/modules/readergesture.lua [hunk 1]
```diff
@@ -686,6 +688,8 @@ function ReaderGesture:buildMenu(ges, default)
         {"suspend", true},
         {"exit", true},
         {"restart", not Device:isAndroid()},
+        {"poweroff", Device:isCervantes() or Device:isKobo() or Device:isSonyPRSTUX()},
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]: With this being duplicated twice here and also in https://github.com/koreader/koreader/blob/master/frontend/ui/elements/common_info_menu_table.lua#L76 maybe it should be abstracted into a Device:canPowerOff() or some such?

Comment (robert00s) [reply frontend/apps/reader/modules/readergesture.lua]: Good tip. Thanks.

**Comment by NiLuJe** [comment]
Sure, it's been used in a couple of places by now ;).

**Comment by poire-z** [comment]
Lgtm.
I'm fine with the abstraction, but we may as well add `canRestart`, which Android can't :)
` {"restart", not Device:isAndroid()},` just above the added ones.

(May be in the menu order, PowerOff after Reboot, ordered by _bye_'ness :) ?)

File: frontend/apps/reader/modules/readergesture.lua [hunk 2]
```diff
@@ -1415,6 +1419,22 @@ function ReaderGesture:gestureAction(action, ges)
...
         self.ui.menu:exitOrRestart(function() UIManager:restartKOReader() end)
+    elseif action == "poweroff" then
+        UIManager:show(ConfirmBox:new{
+            text = _("Are you sure you want to power off the device?"),
+            ok_text = _("Power off"),
+            ok_callback = function()
+                UIManager:nextTick(UIManager.poweroff_action)
+            end,
+        })
+    elseif action == "reboot" then
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readergesture.lua]: Bit of a nitpick that I didn't think of before @poire-z mentioned it, but the reboot/poweroff order also happens here

File: frontend/device/cervantes/device.lua [hunk 3]
```diff
@@ -41,6 +41,8 @@ local Cervantes = Generic:new{
     hasOTAUpdates = yes,
     hasKeys = yes,
     hasWifiManager = yes,
+    canPowerOff = yes,
```
Comment (Frenzie) [inline frontend/device/cervantes/device.lua]: and here (and in the other device files)

Comment (robert00s) [reply frontend/device/cervantes/device.lua]: Done.

**Comment by poire-z** [comment]
> but we may as well add canRestart, which Android can't :)
{"restart", not Device:isAndroid()}, => {"restart", Device:canRestart()}, just above the added ones.

Can you add that one in this PR, for coherency/completeness sake?

**Comment by robert00s** [comment]
Yes, soon.

**Comment by poire-z** [comment]
And some last bits there, and we're done, thanks :)
https://github.com/koreader/koreader/blob/aae6acb48c0b32f1f4898e37441cf7133c12b97b/frontend/apps/reader/modules/readermenu.lua#L211-L215
https://github.com/koreader/koreader/blob/aae6acb48c0b32f1f4898e37441cf7133c12b97b/frontend/apps/filemanager/filemanagermenu.lua#L440-L444

**Pull request merged**